src/bmp_reader.cpp

```diff
@@ -98,20 +111,53 @@ bool ApplyGrayFilter(unsigned char *&header, unsigned char *&rgbData)
 
 bool FlipVertically(unsigned char *&header, unsigned char *&rgbData)
 {
-    const size_t imageSize = *(int *)&header[IMAGE_SIZE_INDEX];
-    const unsigned short bytesPerPixel = (*(short *)&header[BITS_PER_PIXEL_INDEX]) / BITS_PER_BYTE;
-    unsigned int sp = 0;             // start pointer
-    unsigned int ep = imageSize - 1; // end pointer
+    const unsigned int width = *(unsigned int *)&header[WIDTH_INDEX];
```

Comment:
Why did you change the FlipVertically function?

Reply:
Para que se ajustara a los cambios que realice